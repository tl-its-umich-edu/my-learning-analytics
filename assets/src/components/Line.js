import React from 'react'
// import { useDrag } from 'react-dnd'

function Line (props) {
  const {
    height,
    left,
    color,
    width = 2,
    labelPlacement
  } = props

  // const [{ isDragging }, drag] = useDrag({
  //   item: { id: key, type: 'line' },
  //   collect: monitor => ({
  //     isDragging: !!monitor.isDragging()
  //   })
  // })

  const lineHeight = labelPlacement === 'up' || labelPlacement === 'down'
    ? height + 25
    : labelPlacement === 'downLower'
      ? height + 42
      : height + 2

  return (
    <div
      style={{
        position: 'absolute',
        display: 'inline-block',
        width: `${width}px`,
        backgroundColor: color,
        height: `${lineHeight}px`,
        left,
        zIndex: 1,
        marginTop: labelPlacement === 'up' ? '-25px' : 0,
        marginBottom: labelPlacement === 'down' ||
          labelPlacement === 'downLower' ? '-25px' : 0
      }}
      // ref={drag}
    />
  )
}

export default Line
