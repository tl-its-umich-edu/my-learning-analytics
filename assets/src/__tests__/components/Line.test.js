/* global it, expect */

import React from 'react'
import Line from '../../components/Line'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Line
        height={10}
        left={5}
        color='blue'
        labelPlacement='up'
      />
    ).toJSON()

  expect(tree).toMatchSnapshot()
})
