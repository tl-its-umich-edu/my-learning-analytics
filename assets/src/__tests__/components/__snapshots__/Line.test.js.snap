// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "blue",
      "display": "inline-block",
      "height": "35px",
      "left": 5,
      "marginBottom": 0,
      "marginTop": "-25px",
      "position": "absolute",
      "width": "2px",
      "zIndex": 1,
    }
  }
/>
`;
